### ISSUE TITLE
Unable to activate claude sonnet 3.5 new

### ISSUE THREAD

**Issue by UCLCHATA** [microsoft/vscode-copilot-release#2380]
Type: <b>Performance Issue</b>

I have activated/enabled claude from my github copilot setting online but am unable to see them on my copilot chat on the vcsode

Extension version: 0.21.2
VS Code version: Code 1.94.2 (Universal) (384ff7382de624fb94dbaf6da11977bba1ecd427, 2024-10-09T16:08:44.566Z)
OS version: Darwin arm64 23.5.0
Modes:

<details>
<summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|Apple M1 (8 x 2400)|
|GPU Status|2d_canvas: enabled<br>canvas_oop_rasterization: enabled_on<br>direct_rendering_display_compositor: disabled_off_ok<br>gpu_compositing: enabled<br>multiple_raster_threads: enabled_on<br>opengl: enabled_on<br>rasterization: enabled<br>raw_draw: disabled_off_ok<br>skia_graphite: disabled_off<br>video_decode: enabled<br>video_encode: enabled<br>webgl: enabled<br>webgl2: enabled<br>webgpu: enabled<br>webnn: disabled_off|
|Load (avg)|3, 4, 3|
|Memory (System)|8.00GB (0.07GB free)|
|Process Argv|--crash-reporter-id 9282fea8-c8cd-40a7-afe4-6e52c150d932|
|Screen Reader|no|
|VM|0%|
</details><details>
<summary>Process Info</summary>

```
CPU %	Mem MB	   PID	Process
    8	   139	 30783	code main
    0	    57	 30788	   gpu-process
    0	    33	 30789	   utility-network-service
    0	   221	 30794	window [1] (import os.py — Sheldon_6.11_23.42)
    0	    82	 30807	shared-process
    0	     0	 30956	     /bin/ps -ax -o pid=,ppid=,pcpu=,pmem=,command=
    0	    90	 30850	extensionHost [1]
    0	     0	 30876	     /Users/chaya/.vscode/extensions/ms-python.python-2024.18.0-darwin-arm64/python-env-tools/bin/pet server
    0	    49	 30884	     electron-nodejs (server.js )
    0	    49	 30923	     electron-nodejs (bundle.js )
    0	    49	 30851	fileWatcher [1]
    0	    49	 30903	ptyHost
    0	     0	 30904	     /bin/zsh -i
```

</details>
<details>
<summary>Workspace Info</summary>

```
|  Window (import os.py — Sheldon_6.11_23.42)
|    Folder (Sheldon_6.11_23.42): 0 files
|      File types:
|      Conf files:;
```

</details>
<details>
<summary>A/B Experiments</summary>

```
vsliv368:30146709
vspor879:30202332
vspor708:30202333
vspor363:30204092
vscod805cf:30301675
binariesv615:30325510
vsaa593cf:30376535
py29gd2263:31024239
c4g48928:30535728
azure-dev_surveyone:30548225
vscrpc:30673769
962ge761:30959799
pythongtdpath:30769146
pythonnoceb:30805159
asynctok:30898717
pythonmypyd1:30879173
h48ei257:31000450
pythontbext0:30879054
cppperfnew:31000557
dsvsc020:30976470
pythonait:31006305
dsvsc021:30996838
01bff139:31013167
dvdeprecation:31068756
dwnewjupytercf:31046870
impr_priority:31102340
nativerepl2:31139839
refactort:31108082
pythonrstrctxt:31112756
cf971741:31144450
iacca1:31171482
notype1cf:31157160
5fd0e150:31155592
dwcopilot:31170013
g7688163:31175162

```

</details>

**Comment by lramos15** [comment]
Please provide screenshots of your GH settings, VS Code, and the VS Code Copilot Chat output channel (CMD + SHIFT + U -> GitHub Copilot Chat).

**Comment by hindunational** [comment]
PFA the screenshots as asked, let me know if you need anything else.

**Comment by lramos15** [comment]
Emails lose attachments, can you upload them via GitHub

**Comment by elvizlai** [comment]
2024-11-08 07:45:22.439 [info] Using the Electron fetcher.
2024-11-08 07:45:22.439 [info] Initializing Git extension service.
2024-11-08 07:45:22.500 [info] Logged in as elvizlai
2024-11-08 07:45:22.525 [info] Successfully activated the vscode.git extension.
2024-11-08 07:45:22.525 [info] Enablement state of the vscode.git extension: true.
2024-11-08 07:45:22.525 [info] Successfully registered Git commit message provider.
2024-11-08 07:45:24.606 [info] Got Copilot token for elvizlai
2024-11-08 07:45:24.627 [info] copilot token chat_enabled: true, sku: yearly_subscriber
2024-11-08 07:45:24.627 [info] GitHub.vscode-pull-request-github extension is not yet activated.
2024-11-08 07:45:24.630 [info] Registering default platform agent...
2024-11-08 07:45:25.829 [info] Fetched model metadata in 1216ms d8648106-9425-45d1-8bb8-0f0ff86f517f
2024-11-08 07:45:26.003 [info] activationBlocker from 'languageModelAccess' took for 3387ms

Maybe I have similar issue, one of my computer has no choice for claude, but another has that with same account.

**Comment by lramos15** [comment]
@elvizlai Anthropic models are not supported in China. Maybe that is what happening for you as I see your location is in Shanghai.

**Comment by lramos15** [comment]
@hindunational Do you have a VPN you could try just to see if your IP is improperly being grouped as Chinese?

**Comment by hindunational** [comment]
I don't have VPN and im not in China and I'm not at all enjoying using GPT
with copilot, it barely solves any problems. I'm using Claude Sonnet on my
subscription to fix all my error and it's rate limited. So currently I'm
wasting my copilot trial, not enjoying it and likely to not continue.
Please tell me how to get Claude Sonnet working within vscode for me. Btw,
downloaded cursor and it's working fine in their UI, why not yours.

**Comment by lramos15** [comment]
@hindunational So are you stating that you could not enable claude or just you hit a rate limit on Copilot Chat? Just trying to understand the issue at hand.

**Comment by hindunational** [comment]
I could not use Claude at all. I am getting rate limited on using Claude
directly on my own pro account, on vs code it's just not working. I'm
forced to use it on my rate limited pro account because it's not even
showing up as option on my vs code copilot.

**Comment by djensenius** [comment]
@CicerBro the duplicate issue is resolved.

@hindunational Could you try again and let me know?

**Comment by hindunational** [comment]
No, still the same. Only gpt 4o, o1 preview and o1 mini are visible as
options.

**Comment by djensenius** [comment]
@lramos15 I reloaded the users cached settings. How quickly does VS Code refresh the menu options? Can you tell if @hindunational is using the latest VS Code and Plugins?

**Comment by djensenius** [comment]
@hindunational I have given you complimentary access through the end of the year for your bug report. You'll need to go through the signup flow again.

**Comment by lramos15** [comment]
@hindunational Please ensure you update to the latest VS Code as Claude only works there as well. You will want copilot chat 0.22.2 and VS Code 1.95.3

**Comment by be1500616** [comment]
> @hindunational Please ensure you update to the latest VS Code as Claude only works there as well. You will want copilot chat 0.22.2 and VS Code 1.95.3

Thanks for the tip. Updating vs code helped!

**Comment by lramos15** [comment]
We've made many improvements since this issue was filed. Please try again and open a new issue if you're still experiencing problems. This should be resolved.
